contracts/libraries/VaultLifecycleSTETH.sol

```diff
@@ -0,0 +1,747 @@
+            uint64
+        )
+    {
+        return GnosisAuction.placeBid(bidDetails);
+    }
+
+    function claimAuctionOtokens(
+        Vault.AuctionSellOrder calldata auctionSellOrder,
+        address gnosisEasyAuction,
+        address counterpartyThetaVault
+    ) external {
+        GnosisAuction.claimAuctionOtokens(
+            auctionSellOrder,
+            gnosisEasyAuction,
+            counterpartyThetaVault
+        );
+    }
+
+    function verifyConstructorParams(
+        address owner,
+        address feeRecipient,
+        uint256 performanceFee,
+        string calldata tokenName,
+        string calldata tokenSymbol,
+        Vault.VaultParams calldata _vaultParams
+    ) external pure {
+        require(owner != address(0), "!owner");
+        require(feeRecipient != address(0), "!feeRecipient");
+        require(performanceFee > 0, "!performanceFee");
+        require(bytes(tokenName).length > 0, "!tokenName");
+        require(bytes(tokenSymbol).length > 0, "!tokenSymbol");
+
+        require(_vaultParams.asset != address(0), "!asset");
+
+        require(_vaultParams.decimals > 0, "!tokenDecimals");
+        require(_vaultParams.minimumSupply > 0, "!minimumSupply");
+        require(_vaultParams.cap > 0, "!cap");
+    }
+
+    /**
+     * @notice Withdraws yvWETH + WETH (if necessary) from vault using vault shares
+     * @param weth is the weth address
+     * @param asset is the vault asset address
+     * @param collateralToken is the address of the collateral token
+     * @param recipient is the recipient
+     * @param amount is the withdraw amount in `asset`
+     * @return withdrawAmount is the withdraw amount in `collateralToken`
+     */
+    function withdrawYieldAndBaseToken(
+        address weth,
+        address asset,
+        address collateralToken,
+        address recipient,
+        uint256 amount
+    ) external returns (uint256 withdrawAmount) {
+        withdrawAmount = IWSTETH(collateralToken).getWstETHByStETH(amount);
+
+        uint256 yieldTokenBalance =
+            withdrawYieldToken(collateralToken, recipient, withdrawAmount);
+
+        // If there is not enough yvWETH in the vault, it withdraws as much as possible and
+        // transfers the rest in `asset`
+        if (withdrawAmount > yieldTokenBalance) {
+            withdrawBaseToken(
+                weth,
+                asset,
+                collateralToken,
+                recipient,
+                withdrawAmount,
+                yieldTokenBalance
+            );
+        }
+    }
+
+    /**
+     * @notice Withdraws yvWETH from vault
+     * @param collateralToken is the address of the collateral token
+     * @param recipient is the recipient
+     * @param withdrawAmount is the withdraw amount in terms of yearn tokens
+     */
+    function withdrawYieldToken(
+        address collateralToken,
+        address recipient,
+        uint256 withdrawAmount
+    ) internal returns (uint256 yieldTokenBalance) {
```

Comment:
Needs to be external so the ThetaVault contract does not end up importing the code from the library

Reply:
actually needs to be internal because withdrawYieldToken / withdrawBaseToken is used within vault lifecycle methods.

get "withdrawYieldToken" is not (or not yet) visible at this point. otherwise